nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -91,16 +91,6 @@ in {
       '';
     };
 
-    nginx.enable = mkOption {
```

Comment:
The only argument against removing this option is if your `nginx` is running somewhere else i.e. in a different host/container.

Reply:
The workaround would be `systemd.services.nginx.enable = false` in this case.

Reply:
You could still either setup a reverse proxy to the nginx inside the container, or disable it via `mkForce services.nginx.enable = false`. Do you think this is acceptable?

Reply:
I suppose though.

Reply:
But then you would run again into. https://github.com/NixOS/nixpkgs/issues/69261
Therefore it's not really done by switching that one option to false.
I think we need to decide. Either we want to support having nextcloud without nginx (then we need more config options) or we just don't.

Reply:
I would argue that the benefits we get by supporting non-nginx setups aren't big enough to justify the overhead it will have in maintaining it. It just makes things more complicated on every single change. When restricting to nginx in general we can just exclude a lot of question marks whenever we touch anything user/group related
 
Even if you run it inside a container, it shouldn't be a performance bottleneck to have this extra nginx running.

Reply:
Looking for @Ma27 to weigh in on this...

Reply:
I don't need it personally. We could also wait for someone to add it back ;). However it should get deprecated properly like here: https://github.com/NixOS/nixpkgs/blob/6eadc11005318fc4eef7da19898cff51689fb723/nixos/modules/tasks/network-interfaces.nix#L316

Reply:
Will think about this tonight @aanderse (oh an please ping me if I forget about this again :sweat_smile:  )

Reply:
Based on the current implementation, i think it would not be super hard to add support for non-nginx. We just need to add another config option to specify the user of the webserver used. This user will be added to the nextcloud group and phpfpm will be opened to that user.

I already had that implemented before but reverted it. I'm happy to re-revert ;).

The downside of it would be that it makes the module harder to understand for people and therefore harder to maintain i guess.

Reply:
> The downside of it would be that it makes the module harder to understand for people and therefore harder to maintain i guess.

I think I mentioned this in a different thread already, but I might be wrong: I'm rather strongly against supporting multiple reverse-proxies for the exact reasons you gave, even more complexity (and thus more work for the maintainers - including me).

I think that your PR is actually a great solution: we have `nginx` as default (as most of the services here do), but admins now have the freedom to opt-out  of it and use an alternative.